homeassistant/helpers/event.py

```diff
@@ -1609,7 +1609,15 @@ def _interval_listener(self, now: datetime) -> None:
             self._track_job,
             hass.loop.time() + self.seconds,
         )
-        hass.async_run_hass_job(self._run_job, now)
+        if self._run_job.job_type is HassJobType.Coroutinefunction:
+            hass.async_create_periodic_task(
+                # mypy does not know we just checked for the job type
+                self.action(now),  # type: ignore[arg-type]
+                f"track time interval {self.seconds}",
+                eager_start=True,
```

Comment:
Since it's a new job type, should we just eager start all tasks? (as it's the future we want)

Reply:
I made a new runner in 4a1a0e70e8d03db650aeaae8292a8f32d7c49a18. It takes care of the executor case as well now.

So much typing boiler plate ......

Reply:
Its probably better to because I realize that pattern time change listeners and sun listeners probably have the same issue (although less likely since automations don't enable until after startup)

Reply:
Thinking about that some more maybe its not an issue, and I'm not sure if we want to wait for them at shutdown or not so I reverted that for sun/time patterns since people may want their automation to finish and block shutdown

Reply:
Thats still a good question.

If the sun/time happens to hit a state which fires an automation which has to do operation X that takes a few seconds, should it block shutdown?

Reply:
Thinking about it some more, I think it makes sense to cancel them at shutdown as it seems unlikely it should run as its just a likely that it would running while ha is restarting anyways since I don't think people time their shutdowns to avoid specific listeners from firing or when the sun sets

Reply:
```
homeassistant/components/flux_led/__init__.py:    async_track_time_change,
homeassistant/components/flux_led/__init__.py:    entry.async_on_unload(async_track_time_change(hass, _async_sync_time, 3, 40, 30))
homeassistant/components/homeassistant/triggers/time.py:    async_track_time_change,
homeassistant/components/homeassistant/triggers/time.py:                remove = async_track_time_change(
homeassistant/components/homeassistant/triggers/time.py:                async_track_time_change(
homeassistant/components/homeassistant/triggers/time.py:    def remove_track_time_changes() -> None:
homeassistant/components/homeassistant/triggers/time.py:    return remove_track_time_changes
homeassistant/components/homeassistant/triggers/time_pattern.py:from homeassistant.helpers.event import async_track_time_change
homeassistant/components/homeassistant/triggers/time_pattern.py:    return async_track_time_change(
homeassistant/components/pvpc_hourly_pricing/sensor.py:from homeassistant.helpers.event import async_track_time_change
homeassistant/components/pvpc_hourly_pricing/sensor.py:            async_track_time_change(
homeassistant/components/recorder/core.py:    async_track_time_change,
homeassistant/components/recorder/core.py:        self._nightly_listener = async_track_time_change(
homeassistant/components/sonos/switch.py:from homeassistant.helpers.event import async_track_time_change
homeassistant/components/sonos/switch.py:            async_track_time_change(
```

I think all of these make sense to cancel at shutdown so I reverted the revert